### DISCUSSION TITLE
v4.0 - Call for Beta testers

### DISCUSSION THREAD

**Discussion by edwardtfn** [Blackymas/NSPanel_HA_Blueprint#1046] [General]
v4.0 is almost ready and is now on `beta`.
I'd like to invite you to try it in a non-production panel, if you can, and provide us with some feedback. If we have good results from this, I believe we can release in no more than a couple of days from now.

You will find everything you need to install under [`beta`](https://github.com/Blackymas/NSPanel_HA_Blueprint/tree/beta) branch on this repository. [This is a direct link to that](https://github.com/Blackymas/NSPanel_HA_Blueprint/tree/beta).

You can also see here a preview of the [Release Notes](https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/dev/ReleaseNotes.md).

It's not my intention to include any new feature or make any major change before the release. I will focus only on bugs now.
If you get inspired and have a new idea, please create a ["Feature Request" here](https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/new). We can always work on those for the following releases.

If you find some issue that you believe you can help with, please go ahead and propose a change (always in [`dev`](https://github.com/Blackymas/NSPanel_HA_Blueprint/tree/dev) branch, please).
We wanna know even some typo you find in the Release Notes or a minor bug somewhere, so we can take a decision either to fix it or leave for later.

If you don't find anything critical, please drop a message saying we are good to go. 😃 

Thanks a lot for your support.

**Comment by fordcrews** [comment]
Ok, I tried to  replace the url I was using with the one above in one of my panels, and it  couldn't find it.  **repository 'https://github.com/Blackymas/NSPanel_HA_Blueprint/beta/' not found.**   Am I going to have to use replace it with a local copy of the beta yaml instead of loading it locally?   

  remote_package:
    url: https://github.com/Blackymas/NSPanel_HA_Blueprint/tree/beta
    ref: main
    files: [nspanel_esphome.yaml]
    refresh: 300s

I tried without the tree so it was just    url: https://github.com/Blackymas/NSPanel_HA_Blueprint/tree/beta and it couldn't find that either.

**Reply by edwardtfn** [reply]
Use the original url, but replace `ref: main` by `ref: beta`.

Something like this:
```yaml
packages:
  remote_package:
    url: https://github.com/Blackymas/NSPanel_HA_Blueprint
    ref: beta
    files:
      - nspanel_esphome.yaml
    refresh: 300s
```

**Comment by fordcrews** [comment]
Bingo.

**Comment by lordhelmet77** [comment]
I replaced blueprint and nspanel_eu.tft with the beta versions, restarted HA and pushed the TFT update. The blueprint correctly shows 4.0beta. Now the panel is stuck on boot screen:
![IMG_0850](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/124586664/755b727c-fb56-4ae2-a0d5-b28b5d0e9dee)
HA gives me:
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/124586664/af0127d1-fab4-4cd1-bcf6-038243862eea)

Did I miss a step?

**Reply by edwardtfn** [reply]
You have to update ESPHome as indicated [here](https://github.com/Blackymas/NSPanel_HA_Blueprint/discussions/1046#discussioncomment-7039419).

All the 3 parts should be reporting 4.0beta.

Anyways, looks like you have some other issue. Do you have any error message under Home Assistant logs (Settings > System > Logs)?

**Reply by lordhelmet77** [reply]
Yes, my fault. I had the panel flashed to dev-channel. I now went back to dev for blueprint and tft but the panel doesn't boot either.
The log shows:

```2023-09-19 10:00:02.446 ERROR (MainThread) [homeassistant.components.automation.nspanel_dining] NSPanel Dining: Main choices: Automation reloaded: If at step 2: Error executing script. Invalid data for call_service at pos 10: extra keys not allowed @ data['relay1_fallback']
2023-09-19 10:00:02.447 ERROR (MainThread) [homeassistant.components.automation.nspanel_dining] NSPanel Dining: Main choices: Automation reloaded: Error executing script. Invalid data for if at pos 2: extra keys not allowed @ data['relay1_fallback']
2023-09-19 10:00:02.448 ERROR (MainThread) [homeassistant.components.automation.nspanel_dining] NSPanel Dining: Error executing script. Invalid data for choose at pos 1: extra keys not allowed @ data['relay1_fallback']
2023-09-19 10:00:02.449 ERROR (MainThread) [homeassistant.components.automation.nspanel_dining] Error while executing automation automation.nspanel_dining: extra keys not allowed @ data['relay1_fallback']
2023-09-19 10:00:42.560 ERROR (MainThread) [homeassistant.components.automation.nspanel_dining] NSPanel Dining: Main choices: Boot init: If at step 1: Error executing script. Invalid data for call_service at pos 10: extra keys not allowed @ data['relay1_fallback']
2023-09-19 10:00:42.562 ERROR (MainThread) [homeassistant.components.automation.nspanel_dining] NSPanel Dining: Main choices: Boot init: Error executing script. Invalid data for if at pos 1: extra keys not allowed @ data['relay1_fallback']
2023-09-19 10:00:42.563 ERROR (MainThread) [homeassistant.components.automation.nspanel_dining] NSPanel Dining: Error executing script. Invalid data for choose at pos 1: extra keys not allowed @ data['relay1_fallback']
2023-09-19 10:00:42.565 ERROR (MainThread) [homeassistant.components.automation.nspanel_dining] Error while executing automation automation.nspanel_dining: extra keys not allowed @ data['relay1_fallback']```

**Reply by edwardtfn** [reply]
> Invalid data for call_service at pos 10: extra keys not allowed @ data['relay1_fallback']

Are you sure you have flashed the ESPHome from `beta` or `dev` recently? I would expect this error with a ESPHome from a week ago or so, but not with the version there right now.

**Reply by lordhelmet77** [reply]
I flashed it to 'dev' this sunday, two days ago.

So, strange behaviour: I opened ESPhome, opened the device's log … and voilà. Up and running. Also after a reboot. 🎉
After this, I set release channel to beta, swapped blueprint and tft and it works. 🎉🎉

Thanks for your help @edwardtfn! 🎉🎉🎉

**Comment by fordcrews** [comment]
Having the default page the black screen saver, made me think I had hosed my panel at first.

**Reply by edwardtfn** [reply]
How was that? As far as I know, you cannot select `screensaver` as wake-up page. Was that selected in some way?

**Reply by fordcrews** [reply]
After I replaced the old blueprint file with the new one, the logbook on all my panels showed set current page set to screensaver, and they all were black until you touched them.   Not sure why but they go to screensaver now, after ever reboot. 

masternspanel Current page changed to screensaver
10:52:12 AM - 1 minute ago

**Reply by edwardtfn** [reply]
> Not sure why but they go to screensaver now, after ever reboot.

Something is wrong and I wanna investigate this.
After a boot it should go boot>home>screensaver...

How is your "Timeout sleep" set?

**Reply by Doughboy68** [reply]
I noticed this in dev, the default Timeout Sleep on mine was set to a value like 10 or 20 seconds (can't remember exactly). So this puts the panel to sleep by default. If not already since, should this be changed to 0 (to disable sleep) by default because new users may not understand why their panels are going to sleep.

Here is what I have set mine to
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/16811687/45b01270-ff61-4f35-8f41-67557a7ca719)

**Reply by edwardtfn** [reply]
I think I out a very short default on the first try on `dev`, but on the one published to `beta` the defaults are:

Timeout Dimming: 30s
Timeout Page: 15s
Timeout Sleep: 60s

And the timer is reset at the end of the boot process, right after the wake-up page is called, so it should be very similar to the previous behavior on v3.4, where it should sleep around 65s after the end of the boot.
And the code should prevents sleep during the boot.

So, are you sure it went to sleep right after boot? No chance you left the room and by coincidence it went to sleep while you have been out?

**Reply by fordcrews** [reply]
It could have been 60s, I was walking around the house and noticed they were all black, I was used to the not going to sleep.   If they had anything other than the home screen, or the blue screen before the blueprint loads, it has before always been the display update failed.

**Comment by edwardtfn** [comment]
I've just pushed the latest [minor] changes to beta and will wait a few hours. If no one screams, I wanna release v4.0 later today.

**Comment by edwardtfn** [comment]
I'm closing this discussion as v4.0 is released: #1057 

Thanks a lot for your support!

**Comment by edwardtfn** [comment]
I need your support again. I want to push a patch with some fixes, but need your help to run some sanity check so we make sure this is not breaking anything critical.

This is what is changing:

## Overview of all changes

| Change | Criticality | Page/area | Components affected |
| :-- | :--: | :--: | :--: |
| Incorrect button addressing on long press at buttonpage03 | Critical | `buttonpage03` | TFT (US model only) |
| Volume button not working | Medium | `media_player` | TFT |
| Fixes on instructions | Minor | N/A | Blueprint<br>GitHub page (docs) |
| Prevents media player crashing when called before `media_position_updated_at` is able to provide a value. | Minor | `media_player` | Blueprint |
| Add instructions to enforce a timezone selection in customizations | Minor | N/A | GitHub page (docs) |
| Translations to Dutch | Minor | `settings` | Blueprint |
| Translations to Finish (review) | Minor | All | Blueprint |
| Removed `display_current_brightness` messages from log | Minor | N/A | ESPHome |
| Version control to support patches | Minor | N/A | ESPHome |
| Support for flexible time format | Minor | `home` | Blueprint |
| Added date format for Finish | Minor | `home` | Blueprint |
| Reset sleep timer when brightness settings changes | Minor | All | ESPHome |


Thanks again for more this support. 😃

**Comment by edwardtfn** [comment]
Patch released!
https://github.com/Blackymas/NSPanel_HA_Blueprint/releases/tag/v4.0.1

**Comment by edwardtfn** [comment]
v4.1 - Call for beta testers: https://github.com/Blackymas/NSPanel_HA_Blueprint/discussions/1251
